extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,242 @@
+			                                                  case '<':
+				                                                  escaped += "&lt;";
+				                                                  break;
+			                                                  case '>':
+				                                                  escaped += "&gt;";
+				                                                  break;
+			                                                  case '"':
+				                                                  if (input_quote) {
+					                                                  escaped += "&quot;";
+				                                                  } else {
+					                                                  escaped += "\"";
+				                                                  }
+				                                                  break;
+			                                                  case '\'':
+				                                                  if (input_quote) {
+					                                                  escaped += "&#x27;";
+				                                                  } else {
+					                                                  escaped += "\'";
+				                                                  }
+				                                                  break;
+			                                                  default:
+				                                                  escaped += ch;
+			                                                  }
+		                                                  }
+		                                                  return StringVector::AddString(result, escaped);
+	                                                  });
+}
+
+static void add_blob(int &code_point, string &result) {
+	int sz = 0;
+	char c[4] = {'\0', '\0', '\0', '\0'};
+	D_ASSERT(Utf8Proc::CodepointToUtf8(code_point, sz, c));
+	result += c;
+}
+
+static string ReplaceCharref(string &str) {
+	RE2::Options options;
+	options.set_log_errors(false); // disable error logging for performance
+	options.set_case_sensitive(false);
+	const RE2 charref(R"(&(#[0-9]+;?|#[xX][0-9a-fA-F]+;?|[^\t\n\f <&#;]{1,32};?))", options);
+
+	string match;
+	string result {""};
+	int start_pos = 0;
+	duckdb_re2::StringPiece input(str); // wrap a StringPiece around it
+	while (RE2::FindAndConsume(&input, charref, &match)) {
+		// include the input value until the match
+		size_t match_pos = str.find(match, start_pos) - 1; // include the & ampersand charq
+		result.append(str, start_pos, match_pos - start_pos);
+
+		// the position after the end of the current match
+		start_pos = (match_pos + 1) + match.length();
+
+		int num;
+		if (match[0] == '#') {
+			try {
+				// numeric character reference
+				if (match[1] == 'x' || match[1] == 'X') {
+					num = stoi(match.substr(2), nullptr, 16);
+				} else {
+					num = stoi(match.substr(1));
+				}
+			} catch (const std::out_of_range &e) {
+				result += "\uFFFD";
```

Comment:
Generally exceptions are very slow to process and we try to avoid using try/catch inside functions. Perhaps we can use our own internal `TryCast` here? e.g. `TryCast::Operation<string_t, int32_t>()`